pii-detector-service/pii_detector/infrastructure/detector/regex_detector.py

```
@@ -450,22 +450,40 @@ def _is_overlap(self, match1: PIIEntity, match2: PIIEntity) -> bool:
-            )
-        
-        return masked_text
\ No newline at end of file
+            if entity.start < last_pos:
+                continue
+            parts.append(text[last_pos:entity.start])
+            parts.append(f"[{entity.pii_type}]")
+            last_pos = entity.end
+
+        parts.append(text[last_pos:])
+        return "".join(parts)
```

Ruff (0.15.10)

[warning] No newline at end of file

Add trailing newline

(W292)